_data/actions/130.yml

```diff
@@ -17,3 +17,10 @@ desc: |-
       RandomTurn()
   END
   ```
+
+  {% capture note %}
+  * This action will never stop, (unless interrupted in specific situations), and thus never leave the action list.
```

Comment:
This is fishy. So running it once will cause indefinite turning? Quite odd behaviour. Looking at cespenar's script (cespen.baf), he has a chance of running this, but it would break him (only the area ActionOverriding could fix it, but how would it know?).

This script from iwd2 would then not make any sense either:
```
IF
  !TimerActive(1)
THEN
  RESPONSE #100
    RandomTurn()
    StartRandomTimer(1,2,15)
END
```

Reply:
It's in relation to script trigger `ActionListEmpty()`, i.e. `ActionListEmpty()` will never return true.
Anyway, `CESPEN.BCS` contains the following block, which should "release" it from `RandomTurn()`:
```
IF
	Range(Player1,15)
	NumTimesTalkedToGT(0)
	See(Player1)
THEN
	RESPONSE #100
		ClearActions(Myself)
		FaceObject(Player1)
END
```

Dunno about IWD2, guess will probably flag this behavior as EE only...

Reply:
I don't see how this could work. The action is blocking (not in instant.ids), so once it runs, other condition blocks won't be evaluated until it finishes. The break would have to come from a different scriptable.

Reply:
Unless I'm missing something, if a creature is instructed to `RandomTurn()`, it will do that forever.
However, when the script is processed again (from the top of the file), the other condition blocks are evaluated and the corresponding actions will interrupt `RandomTurn()`. It's just that `ActionListEmpty()` will return FALSE...
Anyway, will ask for further details if needed...

Reply:
That's the thing, blocking action don't cause the script to be reprocessed. It's the main reason for the distinction between blocking and instant actions ... ActionListEmpty will of course return false while an action is still running, even if that's not obvious due to internal waits.

You're basically saying that if this was part of the player script, one press of K would get them into endless rotation:
```
IF
	HotKey(K)
THEN
	RESPONSE #100
		RandomTurn()
END
```
I very much doubt it, but I won't have time to test until the evening.

Reply:
Crap, it does cause continuous turning. It's clearly not blocking either, since a True block after it also then triggers all the time. So perhaps they internally mark it as an instant. Which they don't, since this never prints:
```
IF
	HotKey(K)
THEN
	RESPONSE #100
		RandomTurn()
		DisplayString(Myself,3)
END
```
Eh, must be internal logic. So the action is indeed blocking, but the waiting is like with the Wait action, starts script processing from the start.

Reply:
Yeah, it lasts forever but can be interrupted when the script is processed again.
However, as you pointed out, that action must be the last command in an action list (the other actions following it will never be executed).

So, to sum up: should I put an additional note or is the current description fine?

Reply:
I suggest you at least make the "in specific situations" into something more explicit like "with ActionOverride or ClearAllActions".

Reply:
> For pcs any command would do, I guess.

Even for NPCs.
Attach the following script to an NPC and you'll see that it will interrupt `RandomTurn()` as soon as `CHARNAME` is within 10'.

```
IF
	Range(Protagonist,10)
	See(Protagonist)
THEN
	RESPONSE #100
		DisplayStringHead(Myself,3)
		SmallWait(100)
END

IF
	True()
THEN
	RESPONSE #100
		RandomTurn()
END
```